src/components/notifications/NotificationsMenu.js

```diff
@@ -0,0 +1,139 @@
+import React, { useEffect, useState } from "react";
+import { formatDistance, fromUnixTime } from "date-fns";
+import {
+    NOTIFICATIONS_KEY,
+    getNotifications,
+} from "./../../serviceFacades/notifications";
+import { useQuery } from "react-query";
+import { Skeleton } from "@material-ui/lab";
+import NotificationStyles from "./styles";
+import Divider from "@material-ui/core/Divider";
+import ids from "./ids";
+import { build } from "@cyverse-de/ui-lib";
+import Button from "@material-ui/core/Button";
+import {
+    ListItemText,
+    makeStyles,
+    Menu,
+    MenuItem,
+    Typography,
+} from "@material-ui/core";
+import { useTranslation } from "../../i18n";
+
+const useStyles = makeStyles(NotificationStyles);
+
+function getDisplayMessage(notification) {
+    if (notification) {
+        return notification.type === "data"
+            ? notification.subject
+            : notification.message.text;
+    }
+}
+
+function getTimeStamp(time) {
+    if (time) {
+        // slicing because time has extra zeroes in the unix string
+        const d = fromUnixTime(time.slice(0, -3));
+        return formatDistance(d, new Date());
+    }
+}
+
+function NotificationsMenu(props) {
+    const { notificationMssg } = props;
+    const [notifications, setNotifications] = useState([]);
+    const classes = useStyles();
+    const { t } = useTranslation(["common"]);
+
+    const handleClose = () => {
+        props.setAnchorEl();
+    };
+
+    useEffect(() => {
+        if (notificationMssg != null) {
+            setNotifications([notificationMssg, ...notifications].reverse());
+        }
+    }, [notifications, notificationMssg]);
+
+    const { isFetching } = useQuery({
+        queryKey: NOTIFICATIONS_KEY,
+        queryFn: getNotifications,
+        config: {
+            onSuccess: (results) => setNotifications(results?.messages),
+        },
```

Comment:
I noticed in line `53` you were reversing the array to show the latest first. I think that happen here.

Reply:
Also you should update the `unSeenCount` from here so that the `badge` will be shown to user as soon as they login.